schema.graphql

```diff
@@ -934,17 +943,21 @@ type ThawRequest @entity {
 }
 
 """
-A state channel Allocation representing a single Indexer-SubgraphDeployment stake
+A state channel Allocation representing a single Indexer/Provision-SubgraphDeployment stake
 """
 type Allocation @entity {
```

Comment:
we need to add a boolean `isLegacy` to signal wether the allocation was originally created on the `Staking.sol` contract before Horizon, or on the `SubgraphService.sol` contract post horizon

Reply:
Another nice addition would be a `forceClosed` boolean. An allocation is force closed when the msg.sender is not the indexer nor an operator for the indexer. Not sure how complex to implement this would be, if it's too much then don't worry for now.

Reply:
I think we tracked forced closures in another entity before, so it's definitely doable.
Same for the legacy flag.

Reply:
I changed the `AllocationClosed` event to also emit a boolean that specifies wether the allocation was force closed or not.

```
    event AllocationClosed(
        address indexed indexer,
        address indexed allocationId,
        bytes32 indexed subgraphDeploymentId,
        uint256 tokens,
        bool forceClosed
    );
```

Reply:
@tmigone which branch of the contracts repo did you add this to? Trying to generate a new ABI for the subgraph to use with this change.

Reply:
Added it with the same checks we've been already doing for a while, but if the event signature changed, we will need to change it too, so whenever you get a chance to look at this let me know so I can update the latest ABI 😅